backend/infrahub/core/schema/attribute_schema.py

```diff
@@ -53,16 +63,36 @@ def kind_options(cls, v: str) -> str:
 
     @model_validator(mode="before")
     @classmethod
-    def validate_dropdown_choices(cls, values: dict[str, Any]) -> dict[str, Any]:
+    def validate_dropdown_choices(cls, values: Any) -> Any:
         """Validate that choices are defined for a dropdown but not for other kinds."""
-        if values.get("kind") != "Dropdown" and values.get("choices"):
+        if isinstance(values, dict):
+            kind = values.get("kind")
+            choices = values.get("choices")
+        elif isinstance(values, AttributeSchema):
+            kind = values.kind
+            choices = values.choices
+        else:
+            return values
```

Comment:
Will we ever get here? Or will `values` always be either a dict or an AttributeSchema? As I've understood the `mode="before"` part the values will always be converted to a `dict` where as `mode="after"` renders it into the final object where you can reference the fields directly. Do we actually get a `AttributeSchema` object here?

A bit further down we have:
```python
raise ValueError(f"Can only specify 'choices' for kind=Dropdown: {values['kind']}"
```

If the values variable can now be an AttributeSchema the use of `values['kind']` will raise an error we should replace it. with just `kind`as done in the if-statement